### DISCUSSION TITLE
Triggering a synch from an iPhone

### DISCUSSION THREAD

**Discussion by kansaiben1997** [philosowaffle/peloton-to-garmin#380] [Q&A] [answered]
I have everything installed and functioning well on a Mac using the web UI. Wondering how one goes about triggering a synch using a mobile phone though, any guidance on this?

**Comment by manforrilla** [comment]
Bookmark your actions page and then just click your bookmark and tell it to sync

**Reply by kansaiben1997** [reply]
That works for the computer, I’m not clear how I’d use my phone to connect to the web UI though.

**Comment by manforrilla** [comment, marked as answer]
Just use your browser on your phone.
Eric

**Reply by kansaiben1997** [reply]
This may probably be a dumb question but what would the URL be? This is what I use on my computer: http://localhost:8002/ but that’s not something I can access with my phone. Right?

**Reply by philosowaffle** [reply]
You're right, you can't access `localhost` from your phone.  To access the website from your phone you will need to ensure your computer is able to accept external HTTP requests. 

**Most basic option:**
1.  Find the IP of your computer. On your home network this will likely be something like `192.168.1.xx`
2. From your phone trying loading `http://192.168.1.xx:8002`

**Problems you might run into:**
1. Your computer may block the request in order to protect your security
2. You may need to open port `8002` on your computer to allow it to accept the external traffic
3. Your computers IP address may periodically change, meaning the URL you use would also have to change (though there are ways to configure your router to always give your computer the same IP, also known as a static IP)

**Limitations:**
1. You will only be able to access the UI from your home network


Helping set this up is probably beyond the scope of this forum.  There is however a reddit community `r/selfhosted` that is dedicated to self hosting tools like P2G. They also have a [wiki](https://wiki.r-selfhosted.com/getting-started/what-is-self-hosting/) that gets into the details a bit more.

**Reply by kansaiben1997** [reply]
The guidance on the IP address got me where I needed to be. Thanks!

**Comment by manforrilla** [comment]
Awesome, also very sorry, I was assuming you had fork’d in github and where
running the action thru github not on your computer. Thats why I was
wondering why you wouldn’t just log into your account from the browser and
run it.
Eric

**Reply by philosowaffle** [reply]
Ah, didn't even think of that, you've brought up a great option though for any others who stumble across this thread.  

Using the [Github Action](https://philosowaffle.github.io/peloton-to-garmin/install/github-action.html) install method provides a much easier way to access via the browser for those who don't want to get into ipaddresses and ports. [Here](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) are some screenshot instructions for what it looks like.

**Reply by kansaiben1997** [reply]
This was also really helpful guidance as well. I focused in on the Mac setup but the Github Action method is a more versatile option for me. Thanks!
